File: src/Fibonacci.java
/*
Program 4:
    Accept lower and upper limit for the Fibonacci series and print the series
*/

import java.util.Scanner;

public class Fibonacci
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        
        System.out.print("Enter lower and upper limit: ");
        int low = sc.nextInt();
        int up = sc.nextInt();

        if (low >= up)
        {
            System.out.println("Upper limit is too low!");
        }
        else
        {
            System.out.print("The Fibonacci series is: ");
            for (int f1 = 0, f2 = 1, f3; f1 < up; ) // the increment happens inside the body of the loop
            {
                if (f1 >= low)
                    System.out.print(f1 + " ");
                f3 = f1 + f2;
                f1 = f2;
                f2 = f3;
            }
            System.out.println();
        }

        sc.close();
    }
}

File: python/fibonacci.py
# Python script to print Fibonacci series from 'm' to 'n'

start, end = map(int, input("Enter the minimum and maximum range: ").split())

if start >= end:
    print("Range is too small.")
else:
    print("The Fibonacci series is: ", end="")
    f1, f2, f3 = 0, 1, None
    for i in range(0, end + 1):
        if f1 >= start and f1 <= end:
            print(f1, end=" ")
        f3 = f1 + f2
        f1 = f2
        f2 = f3
    print()
